docs/src/main/paradox/server-class.md

```diff
@@ -0,0 +1,117 @@
+
+Akka HTTP provides a [domain-specific language](https://en.wikipedia.org/wiki/Domain-specific_language) (DSL) to simplify the routes/endpoints definition. Each route is composed of one or more `akka.http.scaladsl.server.Directives`, e.g. `path`, `get`, `post`, `complete`, etc.
+
+### Creating a new user
+
+Let us take a look at the source code for the first endpoint, the `/user` URI with a `POST` directive, used to create a new user:
+
+@@snip [QuickstartServer.scala]($g8src$/scala/com/lightbend/akka/http/sample/QuickstartServer.scala) { #user-post }
+
+The snippet above contains a couple of interesting building blocks:
+
+* `path` : matches against the incoming URI, in this case, we want to get all requests that matches `user`.
+* `post` : matches against the incoming Http directive, in this case, we are matching against `POST`.
+* `entity(as[User])` : automatically converts the incoming payload, in this case, we expect JSON, into an entity. We will look more at this functionality in the @ref:[JSON](json.md) section.
+* `complete` : used to reply back to the request. The `StatusCodes.Created` is translated to Http response code 201. We also send back information to the caller in the form of a string.    
+
+When this `Route` is called, we want to create a new user, and we do so by sending a message to the actor `userRegistryActor`. We will look at the implementation of this actor later.
+
+### Retrieving and removing a user
+
+Next we need to define how to retrieve and remove a user, i.e. for the case when the URI `/user/$ID` is used where `$ID` is the id of the user:
+
+@@snip [QuickstartServer.scala]($g8src$/scala/com/lightbend/akka/http/sample/QuickstartServer.scala) { #user-get-delete }
+
+This Route snippet contains a couple of interesting concepts:
+
+* `path("user" / Segment) { => user` : this bit of code matches against URIs of the exact format `/user/$ID` and the `Segment` is automatically extracted into the `user` variable so that we can get to the value passed in the URI. For example `/user/Bruce` will populate the `user` variable with the value "Bruce."
+* `get` : matches against the incoming Http directive.
+
+Let's break down the "business logic" in the first Route:
+
+@@snip [QuickstartServer.scala]($g8src$/scala/com/lightbend/akka/http/sample/QuickstartServer.scala) { #retrieve-user-info }
+
+The code above uses the so-called [ask](http://doc.akka.io/docs/akka/current/scala/actors.html#send-messages) in Akka. This will send a message asynchronously and return a `Future` representing a _possible_ reply. The code above maps the reply to the type `UserInfo`. When the future completes, it will use the second part of the code to evaluate to either `Success`, with or without a result, or a `Failure`. Regardless of the outcome of the future, we should return something to the requester, and we do so by using the `complete` directive with an appropriate response code and value.
```

Comment:
"and return a `Future` representing an eventual reply. The code above maps the reply to the type `Option[UserInfo]`, i.e. either `Some(user)` if it exists or `None` otherwise."

The rest should be rewritten when the more idiomatic route style is used as suggested in the code.